### PR TITLE
Device api

### PR DESCRIPTION
**Pull request by joakim-hove**
Small start of API for device information.

OK: Very basic api usage/example added for devices. You can click on a device on the front page, and come to an information page. Looks like .... - but illustrates the point.

When goint further with API I think we should be strict about using the django url / reverse machinery.

### CODE REVIEW

**Comment by pgdr** [comment]
Is this ready for review?

**Comment by joakim-hove** [comment]
Yes - bring it on!

**Comment by pgdr** [comment]
Aight! :)

File: friskby/templates/friskby/quick.html [hunk 1]
```diff
@@ -72,7 +72,11 @@
       <table class="pure-table">
         <tr><th>Location</th><th>{{pretty_sensor}}</th><th>Last update</th></tr>
           {% for d in device_rows %}
-        <tr><td>{{d.locname}}</td><td class="valuecol">{{d.last}}</td><td>{{d.time}}</td></tr>
+        <tr>
+	  <td><a href="{% url 'view.device.info' pk=d.id %}">{{d.locname}}</a></td>
+	  <td class="valuecol">{{d.last}}</td>
+	  <td>{{d.time}}</td>
+	</tr>
```
Comment (pgdr) [inline friskby/templates/friskby/quick.html]: are these invisible spaces tabs?

File: sensor/api/api_views.py [hunk 2]
```diff
@@ -43,41 +43,88 @@ class DeviceTypeView(generics.RetrieveAPIView):
...
+
+    
 class DeviceView(generics.GenericAPIView , RetrieveModelMixin , UpdateModelMixin):    
     queryset = models.Device.objects.all()
     serializer_class = DeviceSerializer
 
+
+    def get(self, request, *args, **kwargs):
+        if "pk" in kwargs:
+            if "key" in request.GET:
```
Comment (pgdr) [inline sensor/api/api_views.py]: Could be
```python
            key = request.GET['key'] if 'key' in request.GET else None
```

File: sensor/api/api_views.py [hunk 3]
```diff
@@ -43,41 +43,88 @@ class DeviceTypeView(generics.RetrieveAPIView):
...
+        
+    def get_list(self, request):
+        data = []
+        for dev in models.Device.objects.all():
+            location_name = "-------"
+            if dev.location:
+                location_name = dev.location.name
+            data.append( (dev.id , location_name ))
+
+        return Response( data , status = status.HTTP_200_OK )
```
Comment (pgdr) [inline sensor/api/api_views.py]: this function seems like a copy of `DeviceListView.get`?

File: sensor/tests/context.py [hunk 4]
```diff
@@ -14,8 +14,14 @@ def __init__(self):
...
         self.loc = Location.objects.create( name = "Ulriken" , latitude = 200 , longitude = 120 , altitude = 600)
         self.dev_type = DeviceType.objects.create( name = "HP-X123" )
-        self.user = User.objects.create_user( "test-user" )
         self.channel = "stable"
+        self.user_passwd = get_random_string( length = 10 ),
+        self.user = User.objects.create_user( get_random_string( length = 10 ),
+                                              password = self.user_passwd ,
+                                              first_name = "First",
+                                              last_name = "Last",
+                                              email = "joe@invalid.email.com" )
```
Comment (pgdr) [inline sensor/tests/context.py]: could we here have `joe@example.org` instead?  `example.TLD` is meant for such cases.

File: sensor/view/views.py [hunk 5]
```diff
@@ -9,4 +15,13 @@ def get(self , request):
 
 
 
+class DeviceView(View):
+    
+    def get(self, request, pk):
+        device = get_object_or_404( Device, pk = pk)
+
+        # Using the api call:
+        api_get = requests.get( request.build_absolute_uri( reverse("api.device.info" , kwargs = {"pk" : device.id} )))
```
Comment (pgdr) [inline sensor/view/views.py]: I get a rash from long lines, would it be possible to do
```python
        api_get = requests.get(request.build_absolute_uri(reverse("api.device.info",
                                                                  kwargs={"pk": device.id})))
```

File: friskby/templates/friskby/quick.html (hunk 1 above)
Comment (joakim-hove) [inline friskby/templates/friskby/quick.html]: Ehhh - no

File: sensor/api/api_views.py (hunk 3 above)
Comment (joakim-hove) [inline sensor/api/api_views.py]: Yes - `DeviceListView`removed.

File: sensor/tests/context.py (hunk 4 above)
Comment (joakim-hove) [inline sensor/tests/context.py]: Certainly

**Comment by joakim-hove** [comment]
OK - addressed most of the comments; merging.

**Pull request merged**